node/sc/bridge_manager.go

```diff
@@ -589,42 +589,47 @@ func (bm *BridgeManager) RestoreBridges() error {
 	for _, journal := range bm.journal.cache {
 		var cBridgeInfo *BridgeInfo
 		var pBridgeInfo *BridgeInfo
-		var ok bool
+		var cok bool
+		var pok bool
 
 		cBridgeAddr := journal.LocalAddress
 		pBridgeAddr := journal.RemoteAddress
 		bam := bm.subBridge.bridgeAccountManager
 
 		// Set bridge info
-		cBridgeInfo, ok = bm.GetBridgeInfo(cBridgeAddr)
-		if !ok {
+		cBridgeInfo, cok = bm.GetBridgeInfo(cBridgeAddr)
+		pBridgeInfo, pok = bm.GetBridgeInfo(pBridgeAddr)
+		if !cok || !pok {
 			cBridge, err := bridgecontract.NewBridge(cBridgeAddr, bm.subBridge.localBackend)
-			if err != nil || cBridge == nil {
-				logger.Error("local bridge creation is failed", "err", err, "bridge", cBridge)
-				break
-			}
-			err = bm.SetBridgeInfo(cBridgeAddr, cBridge, pBridgeAddr, nil, bam.scAccount, true, false)
 			if err != nil {
-				logger.Error("setting local bridge info is failed", "err", err)
-				bm.DeleteBridgeInfo(cBridgeAddr)
+				logger.Error("local bridge creation is failed", "err", err, "bridge", cBridge)
```

Comment:
If any bridge is failed to connect, the other bridges are useless? Why do you execute `break`?

Reply:
the break stops useless retries. If it fails, chances are high that the peer is disconnected. `RestoreBridges()` retries the remaining bridge subscription after 3 seconds.

Reply:
I think you have an assumption that all bridge addresses are from a single EN. Is this true?
What about any other cases that multiple ENs have different bridge addresses?

Reply:
The `bridge` means `bridge contract` here, not `main bridge` or `sub bridge` module.

Reply:
Okay. for now, I understand this is only for a pair of main-bridge and sub-bridge.